### PR TITLE
Support fallback of xref map

### PR DESCRIPTION
**Pull request by v-pegao**
Support find xref map files in fallback folders.

[Related workitem](https://dev.azure.com/ceapex/Engineering/_workitems/edit/178611/)

Tested the code locally, it can resolve the uid which only defines in fallback xref map files.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5593)

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs [hunk 1]
```diff
@@ -658,5 +659,22 @@ private static XRefSpec GetExternalReference(ExternalReferencePackageCollection
...
         }
+
+        private static IReadOnlyList<string> GetFallbackFolders(ImmutableDictionary<string, object> markdownEngineParameters)
+        {
+            IReadOnlyList<string> fallbackFolders = null;
+            if (markdownEngineParameters.TryGetValue("fallbackFolders", out object obj))
+            {
+                try
+                {
+                    fallbackFolders = ((IEnumerable)obj).Cast<string>().ToList();
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: Can `(List<string>) obj` work?

File: src/Microsoft.DocAsCode.Build.Engine/XRefMaps/XRefMapDownloader.cs [hunk 2]
```diff
@@ -62,6 +65,25 @@ public async Task<IXRefContainer> DownloadAsync(Uri uri)
...
+        {
+            var localFilePath = Path.Combine(_baseFolder, uri.OriginalString);
+            if (File.Exists(localFilePath))
+            {
+                return ReadLocalFile(localFilePath);
+            }
+
+            foreach (var fallbackFolder in _fallbackFolders)
+            {
+                var fallbackFilePath = Path.Combine(fallbackFolder, uri.OriginalString);
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/XRefMaps/XRefMapDownloader.cs]: The logic inside `foreach` is same as `_baseFolder`. What about always put the `_baseFolder` as the first item in `_fallbackFolders`.

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs (hunk 1 above)
Comment (v-pegao) [reply src/Microsoft.DocAsCode.Build.Engine/DocumentBuildContext.cs]: not work, it's object[].

File: src/Microsoft.DocAsCode.Build.Engine/XRefMaps/XRefMapDownloader.cs (hunk 2 above)
Comment (v-pegao) [reply src/Microsoft.DocAsCode.Build.Engine/XRefMaps/XRefMapDownloader.cs]: I merged `_baseFolder` and `_fallbackFolders` to new list in constructor of XRefMapDonwloader

File: src/Microsoft.DocAsCode.Build.Engine/XRefMaps/XRefMapDownloader.cs [hunk 3]
```diff
@@ -16,7 +16,7 @@ public class XRefMapDownloader
     {
         private readonly SemaphoreSlim _semaphore;
         private readonly string _baseFolder;
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/XRefMaps/XRefMapDownloader.cs]: Now we can remove this field.

Comment (v-pegao) [reply src/Microsoft.DocAsCode.Build.Engine/XRefMaps/XRefMapDownloader.cs]: Removed

**Pull request merged**